trustymail/trustymail.py

```diff
@@ -165,71 +166,147 @@ def starttls_scan(domain, smtp_timeout, smtp_localhost, smtp_ports, smtp_cache):
                 # Copy the cached results into the domain object
                 domain.starttls_results[server_and_port] = _SMTP_CACHE[server_and_port]
 
-            
-def spf_scan(resolver, domain):
+
+def check_spf_record(record_text, expected_result, domain):
+    """
+    Test to see if an SPF record is valid and correct.
+
+    The record is tested by checking the response when we query if it
+    allows us to send mail from an IP that is known not to be a mail
+    server that appears in the MX records for ANY domain.
+
+    Parameters
+    ----------
+    record_text : str
+        The text of the SPF record to be tested.
+        
+    expected_result : str
+        The expected result of the test.
+
+    domain : trustymail.Domain 
+        The Domain object corresponding to the SPF record being
+        tested.  Any errors will be logged to this object.
+    """
+    try:
+        # Here I am using the IP address for c1b1.ncats.cyber.dhs.gov
+        # (64.69.57.18) since it (1) has a valid PTR record and (2) is not
+        # listed by anyone as a valid mail server.
+        #
+        # I'm actually temporarily using an IP that virginia.edu resolves to
+        # until we resolve why Google DNS does not return the same PTR records
+        # as the CAL DNS does for 64.69.57.18.
+        query = spf.query("128.143.22.36", "email_wizard@" + domain.domain_name, domain.domain_name, strict=2)
+        response = query.check()
+
+        if response[0] == 'temperror':
+            logging.debug(response[2])
+        elif response[0] == 'permerror':
+            logging.debug('\t' + response[2])
+            domain.syntax_errors.append(response[2])
+        elif response[0] == 'ambiguous':
+            logging.debug('\t' + response[2])
+            domain.syntax_errors.append(response[2])
+        elif response[0] == expected_result:
+            # Everything checks out the SPF syntax seems valid.
+            domain.valid_spf = True
+        else:
+            domain.valid_spf = False
+            logging.debug('\tResult Differs: Expected [{0}] - Actual [{1}]'.format(expected_result, response[0]))
+            domain.errors.append('Result Differs: Expected [{0}] - Actual [{1}]'.format(expected_result, response[0]))
+    except spf.AmbiguityWarning as error:
+        logging.debug('\t' + error.msg)
+        domain.syntax_errors.append(error.msg)
+
+
+def get_spf_record_text(resolver, domain_name, domain, follow_redirect=False):
```

Comment:
Should `follow_redirect` default to True?

Reply:
See my comments below.